### ISSUE TITLE
Dead code in oic.utils.authz.AuthzHandling

### ISSUE THREAD

**Issue by schlenk** [OpenIDC/pyoidc#658]
The authz code does nothing at all right now.

I do not see any calls to the `permissions()` method, and the default call method returns the empty string.

It seems to be intended to help implement `prompt=consent` by returning a list of permissions (e.g. consented scopes) for the session. But the whole other consent machinery is absent, so this does nothing.

**Comment by tpazderka** [comment]
I agree that the structure is somewhat wrong. But we should probably move the `permission` code to the `__call__`.

Our implementation is actually using the `authz` for the consent, I will have a look on Monday as I do not remember the details from top of my head.

**Comment by schlenk** [comment]
Yes. It is called, but does nothing in the code now. The AuthzHandler code seems duplicated, as i think the same checks are already present in the authn code. So yes, we need something to setup session permissions to implement consent, but the current code in the `AuthzHandler` is just a distraction for that.

Curious about consent, where did you hook in the consent requesting UI? authz does not seem to have an option for that, so i would assume it needs to be part of the authn code?

**Comment by tpazderka** [comment]
Our implementation has a consent UI just after the authorization UI. Both sort of handled in our code...

We are using the Auhtz object to store/retrieve consent in our database.

We can probably either remove the code completely as it is unused or we can rewrite the code to be used and provide a sensible default.

I would rather see the latter as an enhancement.

**Comment by schlenk** [comment]
Yes. Some guidance how to do consent and a sensible default implementation would be nice. Better than throwing it away.
